api.lua

```diff
@@ -94,9 +94,13 @@ function areas:canInteract(pos, name)
 	for _, area in pairs(self:getAreasAtPos(pos)) do
 		if area.owner == name or area.open then
 			return true
-		else
-			owned = true
+		elseif areas.factions_available and area.faction_open then
+			local faction_name = factions.get_player_faction(area.owner)
+			if faction_name ~= nil and faction_name == factions.get_player_faction(name) then
```

Comment:
`faction_name ~= nil` is possibly not needed. You can also compare nil with string, and it'll be `false`.

Reply:
It is needed, as `get_player_faction` can return `nil` as well if the player is not in a faction.